libs/langchain-community/src/graphs/graph_document.ts

```diff
@@ -0,0 +1,86 @@
+import { Serializable } from "@langchain/core/load/serializable";
+import { Document } from "@langchain/core/documents";
+
+export class Node extends Serializable {
+  id: string | number;
+
+  type: string;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  properties: Record<string, any>;
+
+  lc_namespace = ["langchain", "graph", "document_node"];
+
+  constructor({
+    id,
+    type = "Node",
+    properties = {},
+  }: {
+    id: string | number;
+    type: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    properties?: Record<string, any>;
+  }) {
+    super();
+    this.id = id;
+    this.type = type;
+    this.properties = properties;
+  }
+}
+
+export class Relationship extends Serializable {
+  source: Node;
+
+  target: Node;
+
+  type: string;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  properties: Record<string, any>;
+
+  lc_namespace = ["langchain", "graph", "document_relationship"];
+
+  constructor({
+    source,
+    target,
+    type,
+    properties = {},
+  }: {
+    source: Node;
+    target: Node;
+    type: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    properties?: Record<string, any>;
+  }) {
+    super();
+    this.source = source;
+    this.target = target;
+    this.type = type;
+    this.properties = properties;
+  }
+}
+
+export class GraphDocument extends Document {
+  nodes: Node[];
+
+  relationships: Relationship[];
+
+  source: Document;
```

Comment:
Unnest this since we're now extending `Document` directly? Otherwise there's no point

Reply:
Basically implement `pageContent` and `metadata`

Reply:
Would consisteny between python and JS be a higher priority? IMO, there is not much value in extending the doc for rag, as we never use the graph doc in retrieval setting, it's only used for ingestion atm. Even if we wanted to use it in retrieval, it would need custom components since none use nodes or rels atm

Reply:
Ok, fine by me.